test/example-types/facet.js

```diff
@@ -507,27 +520,61 @@ describe('facet', () => {
         let ids = _.map(({ name }) => _.toString(name), result.options)
         expect(_.includes('5e9dbd76e991760021124966', ids)).to.be.true
       })
-      it('when the first and second search result does not  contain the checked value', async () => {
+      it('when the first and second search results do not  contain the checked value', async () => {
         let mockConfig = {
           getProvider: () => ({
             runSearch: () => [{ label: { name: '5' }, _id: 5 }],
           }),
           getSchema() {},
         }
-
-        node.label.collection = Data
+        node.field = '_id'
+        node.label.collection = mongoIdData
         node.isMongoId = null
         node.values = [5]
         let result = await facet.result(
           node,
-          agg => mingo.aggregate(Data, agg),
+          agg => mingo.aggregate(mongoIdData, agg),
           {},
           mockConfig
         )
         let ids = _.map(({ name }) => _.toString(name), result.options)
         expect(_.includes('5', ids)).to.be.true
       })
-      it('when the first and second search result does not  contain the checked value  and  isMongoId is true', async () => {
+      it('when the first and second search results do not contain the checked value and values are boolean', async () => {
+        let mockConfig = {
+          getProvider: () => ({
+            runSearch: () => [{ label: { bool: true }, _id: 5 }],
+          }),
+          getSchema() {},
+        }
+        node.field = 'bool'
+        node.label.collection = mongoIdData
+        node.isMongoId = null
+        node.values = [true]
+        let result = await facet.result(node, () => [], {}, mockConfig)
+
+        let ids = _.map(({ name }) => _.toString(name), result.options)
+        expect(_.includes('5', ids)).to.be.true
+      })
+      it('when the field label.fields is undefined', async () => {
+        let mockConfig = {
+          getProvider: () => ({
+            runSearch: () => [{ label: { bool: true }, _id: 5 }],
+          }),
+          getSchema() {},
+        }
+        node.size = 1
+        node.field = 'bool'
+        ;(node.label = {
+          collection: mongoIdData,
+          foreignField: '_id',
+        }),
+          (node.values = [true])
```

Comment:
what is this weird syntax? 🤔

Reply:
Duti changed my code to this weird format. cause I had an extra comma